packages/react-native-passkey-stamper/src/index.ts

```diff
@@ -0,0 +1,155 @@
+/// <reference lib="dom" />
+import { Passkey } from "react-native-passkey";
+import { sha256 } from "@noble/hashes/sha256";
+import type { TurnkeyApiTypes } from "@turnkey/http";
+
+// https://www.w3.org/TR/webauthn-2/#dictionary-credential-descriptor
+// Copied from https://github.com/f-23/react-native-passkey/blob/17184a1b1f6f3ac61e07aa784c9b64efb28b570e/src/Passkey.tsx#L80C1-L85C2
+// TODO: can we import this type instead?
+interface PublicKeyCredentialDescriptor {
+  type: string;
+  id: string;
+  transports?: Array<string>;
+}
+
+export type TurnkeyAuthenticatorParams =
+  TurnkeyApiTypes["v1AuthenticatorParamsV2"];
+
+// Header name for a webauthn stamp
+const stampHeaderName = "X-Stamp-Webauthn";
+
+export type TPasskeyRegistrationConfig = {
+  // The RPID ("Relying Party ID") for your app.
+  // See https://github.com/f-23/react-native-passkey?tab=readme-ov-file#configuration to set this up.
+  rpId: string;
+  rpName: string;
+  // Properties for passkey display: user name and email will show up in the prompts
+  user: {
+    id: string;
+    name: string;
+    displayName: string;
+  };
+  // Name of the authenticator (affects Turnkey only, won't be shown on passkey prompts)
+  // TODO: document restrictions on character sets
+  authenticatorName: string;
+  // Optional challenge. If not provided, a new random challenge will be generated
+  challenge?: string;
+  // Optional timeout value. Defaults to 5 minutes.
+  timeout?: number;
+  // Optional override for UV flag. Defaults to "preferred".
+  userVerification?: UserVerificationRequirement;
+  // Optional list of credentials to pass. Defaults to empty
+  allowCredentials?: PublicKeyCredentialDescriptor[];
+};
+
+export type TPasskeyStamperConfig = {
+  // The RPID ("Relying Party ID") for your app.
+  // See https://github.com/f-23/react-native-passkey?tab=readme-ov-file#configuration to set this up.
+  rpId: string;
+  // Optional timeout value. Defaults to 5 minutes.
+  timeout?: number;
+  // Optional override for UV flag. Defaults to "preferred".
+  userVerification?: UserVerificationRequirement;
+  // Optional list of credentials to pass. Defaults to empty
+  allowCredentials?: PublicKeyCredentialDescriptor[];
+};
+
+const defaultTimeout = 5 * 60 * 1000; // five minutes
+const defaultUserVerification = "preferred";
+
+/**
+ * Creates a passkey and returns authenticator params
+ */
+export async function CreatePasskey(
+  config: TPasskeyRegistrationConfig
+): Promise<TurnkeyAuthenticatorParams> {
+  const arr = new Uint8Array(32);
+  crypto.getRandomValues(arr);
+  // TODO: this is probably not right? Need to find the right way to encode random bytes into a string
+  const challenge = new TextDecoder("utf-8").decode(arr);
```

Comment:
- Seems like `crypto` and `TextDecoder` is not available in react-native (testing on android emulator).

Reply:
Ah, we just need some polyfill to get this working. Maybe could add to readme

### crypto
1. `yarn add react-native-get-random-values`
2. On react-native versions below 0.6 you need to do this `react-native link react-native-get-random-values`
3. `import 'react-native-get-random-values';`  at the top of your entry file (e.g., index.js):

### TextDecoder
1. yarn add text-encoding
2. Add this to your entry point, e.g., index.js
```
import { TextDecoder, TextEncoder } from 'text-encoding';

global.TextDecoder = global.TextDecoder || TextDecoder;
global.TextEncoder = global.TextEncoder || TextEncoder;
```

Reply:
Thanks for looking @oskarleonard! I think it makes sense to load polyfills for textEncoder / crypto.getRandomValues

@Balanced02 let's pull this in a different file, you're right! 👍

Reply:
We added these polyfills for our app, so that is no problem for us anymore. Just a FYI, that this package should probably add these instructions to a readme. 

What i mean to say, i dont know if it is best that the polyfills are in this package or if the apps using this package should need to add those polyfills to make this package work.

Reply:
Addressed in https://github.com/tkhq/sdk/pull/209/commits/0a15f9e757382ab9b7aa058d27481c78672c7c5a